torch/distributed/distributed_c10d.py

```diff
@@ -2209,11 +2209,18 @@ def _object_to_tensor(obj, device):
     # Otherwise, it will casue 100X slowdown.
     # See: https://github.com/pytorch/pytorch/issues/65696
     byte_tensor = torch.ByteTensor(byte_storage).to(device)
+    if get_debug_level() == DebugLevel.DETAIL and is_nccl_available():
+        hash = torch._C._distributed_c10d._hash_tensors([byte_tensor])
+        logger.warning(f"_object_to_tensor size: {byte_tensor.numel()} hash value: {hash}")  # noqa: G004
     local_size = torch.LongTensor([byte_tensor.numel()]).to(device)
     return byte_tensor, local_size
 
 
 def _tensor_to_object(tensor, tensor_size):
+    if get_debug_level() == DebugLevel.DETAIL and is_nccl_available():
+        cuda_tensor = tensor.cuda()
```

Comment:
Instead of moving to cuda(), should we just test if the tensor is on GPU? If the tensor is on CPU, it is NOT using NCCL to do the communication.